src/Cake.Core/ICakeTaskInfo.cs

```diff
@@ -27,6 +27,6 @@ public interface ICakeTaskInfo
         /// Gets the task's dependencies.
         /// </summary>
         /// <value>The task's dependencies.</value>
-        IReadOnlyList<string> Dependencies { get; }
+        IReadOnlyList<CakeTaskDependency> Dependencies { get; }
```

Comment:
Yes, I agree. There are some things planned for 0.19 that addresses task dependencies, so we should wait with this until we've had the chance to have a proper discussion about that as well before introducing something new.